h/static/scripts/group-forms/components/CreateGroupForm.tsx

```diff
@@ -42,36 +47,80 @@ function Label({
   );
 }
 
-export default function CreateGroupForm() {
-  const nameId = useId();
-  const descriptionId = useId();
+function TextField({
+  type,
+  limit,
+  label,
+  testid,
+  required = false,
+  autofocus = false,
+  classes = '',
+}: {
+  type: 'input' | 'textarea';
+  limit: number;
+  label: string;
+  testid: string;
+  required?: boolean;
+  autofocus?: boolean;
+  classes?: string;
+}) {
+  const id = useId();
+  const [value, setValue] = useState('');
+
+  function handleInput(e: InputEvent) {
+    setValue((e.target as HTMLInputElement).value);
+  }
+
+  const error = value.length > limit;
 
+  const InputComponent = type === 'input' ? Input : Textarea;
+
+  return (
+    <div className="mb-4">
+      <Label htmlFor={id} text={label} required={required} />
+      <InputComponent
+        id={id}
+        onInput={handleInput}
+        error={error ? `Must be ${limit} characters or less.` : undefined}
+        value={value}
+        classes={classes}
+        autofocus={autofocus}
+        autocomplete="off"
+        data-testid={testid}
+      />
+      <CharacterCounter
+        value={value.length}
+        limit={limit}
+        testid={`charcounter-${testid}`}
+        error={error}
+      />
```

Comment:
I notice the character count shifts down lightly when it changes from being under to over the limit. We might need to tweak the spacing or line-height slightly to avoid this or make it more subtle.

Reply:
I'm actually not seeing it shift down in either Firefox or Chrome on Linux on my screen. I wonder if it's a macOS thing? I briefly tried testing on my MacBook but it has that issue with Elasticsearch not booting up. I've made a note of this on the project board to come back to it later.

[Screencast from 2024-07-17 10-12-46.webm](https://github.com/user-attachments/assets/d9be6119-583a-4d73-bcfd-90121efcd2e5)

Reply:
I can see the issue in macOS but only in Safari. Had to increase the `line-height` all the way up to `1.625` (`leading-relaxed`) to fix it, but that's probably fine.